api/application.go

```diff
@@ -596,7 +620,11 @@ func PurgeAccountStorage(context DatabaseService, account int) {
 
 // Hash returns the SHA256 hash of the application state in hexadecimal
 func Hash(context DatabaseService, account int) string {
-	hash := sha256.Sum256(Application(context, account, true))
+	jsonBytes, err := Application(context, account, true)
+	if err != nil {
+		return ""
```

Comment:
This only seems marginally better than just continuing to send a (bad) hash in the event of an error. There's no indication via function signature that the caller needs to check for an error. And it looks like while one call point was updated to check for the empty string, there is at least another where it was not updated.

This change motivated my questions about recover/panic and if there was a way to just catch/log these type of errors at the top-level HTTP request handler vs having to check them at every call up the stack.

Reply:
I think that using panic is possible, but is not common at all in go code. Go is pedantic in this way but very "correct". I also didn't really like that this was returning an empty string instead of an error, I'll make it an explicit err like everything else.